File: vane-proxy-core/src/main/java/org/oddlama/vane/proxycore/ProxyServer.java
package org.oddlama.vane.proxycore;

import java.util.Collection;
import java.util.UUID;
import org.oddlama.vane.proxycore.scheduler.ProxyTaskScheduler;

public interface ProxyServer {
    ProxyTaskScheduler getScheduler();

    void broadcast(String message);

    Collection<ProxyPlayer> getPlayers();

    boolean hasPermission(UUID uuid, String... permission);
}

File: vane-proxy-core/src/main/kotlin/org/oddlama/vane/proxycore/ProxyServer.kt
package org.oddlama.vane.proxycore

import org.oddlama.vane.proxycore.scheduler.ProxyTaskScheduler
import java.util.*

interface ProxyServer {
    val scheduler: ProxyTaskScheduler?

    fun broadcast(message: String?)

    val players: MutableCollection<ProxyPlayer?>?

    fun hasPermission(uuid: UUID?, vararg permission: String?): Boolean
}
